includes/multi-currency/StorefrontIntegration.php

```diff
@@ -0,0 +1,154 @@
+<?php
+/**
+ * Class StorefrontIntegration
+ *
+ * @package WooCommerce\Payments\StorefrontIntegration
+ */
+
+namespace WCPay\MultiCurrency;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class that controls Multi Currency Storefront Integration.
+ */
+class StorefrontIntegration {
+	/**
+	 * The plugin's ID.
+	 *
+	 * @var string
+	 */
+	public $id;
+
+	/**
+	 * Multi-Currency instance.
+	 *
+	 * @var MultiCurrency
+	 */
+	protected $multi_currency;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param MultiCurrency $multi_currency The MultiCurrency instance.
+	 */
+	public function __construct( MultiCurrency $multi_currency ) {
+		$this->multi_currency = $multi_currency;
+		$this->id             = $this->multi_currency->id;
+		$this->init_actions_and_filters();
+	}
+
+	/**
+	 * Adds the checkbox to the store settings to toggle the currency switched in the Storefront breadcrumb section.
+	 *
+	 * @param array $settings The settings from the Settings class.
+	 *
+	 * @return array The modified settings array.
+	 */
+	public function filter_store_settings( array $settings ): array {
+		foreach ( $settings as $key => $value ) {
+			if ( isset( $value['id'] ) ) {
```

Comment:
Can we decrease the levels of indentation here? One approach would be to use continue:

```
if ( ! isset( $value['id'] ) ) {
   continue;
}

```

https://www.php.net/manual/en/control-structures.continue.php

Reply:
Thanks for pointing it out @dwainm! I always prefer to this "early break" (continue in this case) behavior to keep the real login just below the guards/checkers. And missed this one!
_Addressed in 691e819_